crates/bevy_gizmos/src/primitives/dim2.rs

```diff
@@ -0,0 +1,434 @@
+
+use super::helpers::*;
+
+use bevy_math::primitives::{
+    BoxedPolygon, BoxedPolyline2d, Circle, Direction2d, Ellipse, Line2d, Plane2d, Polygon,
+    Polyline2d, Primitive2d, Rectangle, RegularPolygon, Segment2d, Triangle2d,
+};
+use bevy_math::{Mat2, Vec2};
+use bevy_render::color::Color;
+
+use crate::prelude::{GizmoConfigGroup, Gizmos};
+
+/// A trait for rendering 2D geometric primitives (`P`) with [`Gizmos`].
+pub trait GizmoPrimitive2d<P: Primitive2d> {
+    /// The output of `primitive_2d`. This is a builder to set non-default values.
+    type Output<'a>
+    where
+        Self: 'a;
+
+    /// Renders a 2D primitive with its associated details.
+    fn primitive_2d(
+        &mut self,
+        primitive: P,
+        position: Vec2,
+        rotation: Mat2,
+        color: Color,
+    ) -> Self::Output<'_>;
+}
+
+// direction 2d
+
+impl<'w, 's, T: GizmoConfigGroup> GizmoPrimitive2d<Direction2d> for Gizmos<'w, 's, T> {
+    type Output<'a> = () where Self : 'a;
+
+    fn primitive_2d(
+        &mut self,
+        primitive: Direction2d,
+        position: Vec2,
+        rotation: Mat2,
+        color: Color,
+    ) -> Self::Output<'_> {
+        if !self.enabled {
+            return;
+        }
+
+        let start = position;
+        let end = position + (rotation * *primitive);
+        self.arrow_2d(start, end, color);
+    }
+}
+
+// circle 2d
+
+impl<'w, 's, T: GizmoConfigGroup> GizmoPrimitive2d<Circle> for Gizmos<'w, 's, T> {
+    type Output<'a> = () where Self: 'a;
+
+    fn primitive_2d(
+        &mut self,
+        primitive: Circle,
+        position: Vec2,
+        _rotation: Mat2,
+        color: Color,
+    ) -> Self::Output<'_> {
+        if !self.enabled {
+            return;
+        }
+
+        self.circle_2d(position, primitive.radius, color);
+    }
+}
+
+// ellipse 2d
+
+impl<'w, 's, T: GizmoConfigGroup> GizmoPrimitive2d<Ellipse> for Gizmos<'w, 's, T> {
+    type Output<'a> = () where Self: 'a;
+
+    fn primitive_2d(
+        &mut self,
+        primitive: Ellipse,
+        position: Vec2,
+        rotation: Mat2,
+        color: Color,
+    ) -> Self::Output<'_> {
+        if !self.enabled {
+            return;
+        }
+
+        self.ellipse_2d(
+            position,
+            rotation,
+            primitive.half_size.x,
+            primitive.half_size.y,
+            color,
+        );
+    }
+}
+
+// line 2d
+
+impl<'w, 's, T: GizmoConfigGroup> GizmoPrimitive2d<Line2d> for Gizmos<'w, 's, T> {
+    type Output<'a> = () where Self: 'a;
+
+    fn primitive_2d(
+        &mut self,
+        primitive: Line2d,
+        position: Vec2,
+        rotation: Mat2,
+        color: Color,
+    ) -> Self::Output<'_> {
+        if !self.enabled {
+            return;
+        }
+
+        let direction = rotation * *primitive.direction;
+
+        let start = position;
+        let end = position + direction;
+        self.arrow_2d(start, end, color);
```

Comment:
No strong feelings here, but it might make more sense to have the arrowhead at the center of the line, at `position`
```suggestion
        let start = position - direction;
        let end = position;
        self.arrow_2d(start, end, color);
```

Reply:
I think I agree almost everywhere. An exception is here. It looks a bit weird to me personally. I tried out both:

### Old

![image](https://github.com/bevyengine/bevy/assets/26892280/00f5380c-20c7-47da-b09d-819d0fee21ce)


### New

![image](https://github.com/bevyengine/bevy/assets/26892280/1e3c69aa-7ff0-4c3e-b433-f7404ea27eb1)